### PR TITLE
Show/Hide events table columns

### PR DESCRIPTION
**Pull request by hriday-panchasara**
- Added an optional 'Options menu button' in AbstractOutputComponent's Title Bar:
Screen Shot 2022-04-14 at 3 47 48 PM

- Events table implements this button to open a dropdown list that allows users to toggle visibility of multiple columns
Screen Shot 2022-04-14 at 3 47 14 PM

### CODE REVIEW

File: packages/react-components/style/output-components-style.css [hunk 1]
```diff
@@ -248,3 +253,21 @@ canvas {
 .hoverClass:hover {
     border: 1px solid black;
 }
+
+.options-menu-drop-down {
+    position: absolute;
```
Comment (PatrickTasse) [inline packages/react-components/style/output-components-style.css:258]: It seems to me that as a menu popup, the dialog should appear in a position relative to the mouse cursor or the menu icon.

Especially that where it is placed now, and with no user indication of how to close the dialog (click outside?), the user might mistakenly think that the [x] close button is to close the dialog, but it closes the table output.

File: packages/react-components/src/components/abstract-output-component.tsx [hunk 2]
```diff
@@ -166,7 +166,7 @@ export abstract class AbstractOutputComponent<P extends AbstractOutputProps, S e
     }
 
     private closeOptionsMenu(event: Event): void {
-        if (event.currentTarget instanceof Node && this.optionsMenuRef.current?.contains(event.currentTarget)) {
+        if (event.target instanceof Node && this.optionsMenuRef.current?.contains(event.target)) {
```
Comment (PatrickTasse) [inline packages/react-components/src/components/abstract-output-component.tsx:169]: This change belongs in the first commit?

File: packages/react-components/style/output-components-style.css [hunk 3]
```diff
@@ -257,15 +257,15 @@ canvas {
 .options-menu-drop-down {
     position: absolute;
     left: 100%;
-    width: 300px;
+    width: 250px;
```
Comment (PatrickTasse) [inline packages/react-components/style/output-components-style.css:260]: These changes belong in the first commit?

File: packages/react-components/src/components/abstract-output-component.tsx [hunk 4]
```diff
@@ -94,7 +100,11 @@ export abstract class AbstractOutputComponent<P extends AbstractOutputProps, S e
                 {outputName}
                 <i id={this.props.traceId + this.props.outputDescriptor.id + 'handleSpinner'} className='fa fa-refresh fa-spin'
                     style={{ marginTop: '5px', visibility: 'hidden'}} />
+                {this.state.optionsDropdownOpen !== undefined && <button title='Show View Options' className='remove-component-button' onClick={this.openOptionsMenu}>
+                    <FontAwesomeIcon icon={faBars} />
```
Comment (PatrickTasse) [inline packages/react-components/src/components/abstract-output-component.tsx:104]: I think the options menu button should be snapped below the close button, not floating when output height is resized.

File: packages/react-components/src/components/table-output-component.tsx [hunk 5]
```diff
@@ -653,4 +654,56 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
...
+        const columns = this.columnApi.getAllColumns();
+
+        return <React.Fragment>
+            <table style={{width: '100%'}}>
+                <thead>
+                    <tr style={{textAlign: 'left'}}>
+                        <th>Show/Hide</th>
+                        <th>Column Name</th>
+                    </tr>
+                </thead>
```
Comment (PatrickTasse) [inline packages/react-components/src/components/table-output-component.tsx:691]: In my opinion, this table header is unnecessary and I would prefer the checkboxes much closer to the column names.

**Review by PatrickTasse — CHANGES_REQUESTED** [review]

**Comment by hriday-panchasara** [comment]
@PatrickTasse Changes Made:
- Rebased so all changes belong to one commit
- Removed table header
- Options menu snapped below the close button. Clicking on this would open a dropdown menu. Dropdown menu has 'Toggle Columns' option - when clicked the table will appear. The idea is to have this dropdown also include 'Export View' and 'Pin View' options. I'll add them to this dropdown once this item has merged and I'll have to resolve merge conflicts in those PR's
Screen Shot 2022-04-22 at 4 25 25 PM
- Dropdown appears relative to menu icon

**Review by PatrickTasse — APPROVED** [review]

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks for this feature.

**Pull request merged**